### ISSUE TITLE
How to un-set a default value for ParameterSpec.Builder?

### ISSUE THREAD

**Issue by mjburghoffer** [square/kotlinpoet#925]
I am trying to re-use an existing `ParameterSpec` like so:

```
val strictParam = param.toBuilder().defaultValue(null)
// or
val strictParam2 = param.toBuilder().removeDefault()
// etc
```

The first method does not compile, due to the input being non-nullable.  The second method doesn't exist.  Is there some other undocumented feature that I am missing?

Thank you!

**Comment by mjburghoffer** [comment]
I suppose - can just copy the old version as a workaround:
```
val strictParam = ParameterSpec(param.name, param.type, param.modifiers)
```

**Comment by Egorand** [comment]
Yeah, we don't provide any methods for unsetting a default value in `ParameterSpec.Builder`. Perhaps we could add one, or expose the field so it can be mutated directly - we've done it for a few other builders.

For the time being, consider reusing a `ParameterSpec.Builder` without the default value in your code, and only set the default value in cases when it's needed, hopefully that makes sense.

**Comment by Egorand** [comment]
@Dominaezzz if it's possible for the code using KotlinPoet to not set default values on `actual` functions that'd be the best solution. I don't think ignoring is the right approach here, if we added code to check for this case I think we'd crash instead.
